rudder-web/src/main/scala/com/normation/rudder/web/rest/RestDataSerializer.scala

```diff
@@ -586,4 +590,36 @@ case class RestDataSerializerImpl (
     )
   }
 
+  def serializeDataSource(source : DataSource): JValue = {
+    ( ( "name"        -> source.name.value  )
+    ~ ( "id"        -> source.id.value  )
+    ~ ( "description" -> source.description )
+    ~ ( "type" -> (
+        ( "name" -> source.sourceType.name )
+        ~ { source.sourceType match {
+          case HttpDataSourceType(url,headers,method,path,mode,timeOut) =>
+            ( ( "url"        -> url     )
+            ~ ( "headers"    -> headers )
+            ~ ( "path"       -> path    )
+            ~ ( "requestTimeout"  -> timeOut.length )
```

Comment:
length is not what we want: 
- it is the number of Unit (but not the unit)
- it is not defined for not finite duration. 

So either we use here a ".toStrign", which seems to be consistant with the unapply of the different duration (finite or not), and it means that we will have things like "102 ms" and "Inf" in the JSON, or we always transform to something (ms, likelly), and have a norm for infinite (Long.[Min|Max]Value, most likelly), and we have some logic to handle these special number in both way.